paimon-api/src/main/java/org/apache/paimon/rest/RESTApi.java

```diff
@@ -851,6 +852,7 @@ public List<String> listFunctions(String databaseName) {
      * @throws ForbiddenException if the user lacks permission to access the function
      */
     public GetFunctionResponse getFunction(Identifier identifier) {
+        FunctionNameValidator.check(identifier.getObjectName());
```

Comment:
if (!isValidName(name)) {
   throw new NoSuchResourceException
}

Reply:
If using NoSuchResourceException, this exception is not reasonable when creating a function. So here, IllegalArgumentException may be better. We could give different in function's api in catalog.